omnigibson/controllers/controller_base.py

```diff
@@ -372,15 +364,23 @@ def nums2array(nums, dim):
             dim (int): Size of array to broadcast input to
 
         Returns:
-            np.array: Array filled with values specified in @nums
+            th.tensor: Array filled with values specified in @nums
         """
         # First run sanity check to make sure no strings are being inputted
         if isinstance(nums, str):
             raise TypeError("Error: Only numeric inputs are supported for this function, nums2array!")
 
-        # Check if input is an Iterable, if so, we simply convert the input to np.array and return
+        # Check if input is an Iterable, if so, we simply convert the input to th.tensor and return
         # Else, input is a single value, so we map to a numpy array of correct size and return
-        return np.array(nums) if isinstance(nums, Iterable) else np.ones(dim) * nums
+        return (
+            nums.float()
+            if isinstance(nums, th.Tensor)
```

Comment:
why is it necessary to call `nums.float()`? In general, I felt like we should always use float32 for torch tensors, unless there is a strong need to use float64?